scripts/firedrake-install

```diff
@@ -520,21 +520,31 @@ def sniff_compiler(exe):
     # VERSION
     try:
         output = subprocess.run(
-            [exe, "-dumpversion"],
+            [exe, "-dumpfullversion"],
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             check=True,
             encoding="utf-8"
         ).stdout
         version = Version(output)
     except (subprocess.CalledProcessError, UnicodeDecodeError, InvalidVersion):
-        version = None
+        try:
```

Comment:
This is quite an unpleasant bit of code. A refactor of this might be
```py
for dumpstr in ["-dumpversion", "-dumpfullversion"]:
    try:
        output = subprocess.run(...)
        version = ...
        break
    except (...):
        continue
```
Less code but arguably less clear. WDYT?